diff -ruN VTK-9.6.0/Infovis/Core/vtkWordCloud.cxx VTK-9.6.0.patched/Infovis/Core/vtkWordCloud.cxx
--- VTK-9.6.0/Infovis/Core/vtkWordCloud.cxx	2026-02-11 04:19:44.000000000 +0100
+++ VTK-9.6.0.patched/Infovis/Core/vtkWordCloud.cxx	2026-03-11 08:33:20.442916533 +0100
@@ -652,9 +652,9 @@
   const int centerX = sizes[0] / 2.0;
   const int centerY = sizes[1] / 2.0;
 
-  constexpr std::size_t N = 10000;
+  const std::size_t N = 10000;
   constexpr auto pi = 3.141592653589793238462643383279502884L; /* pi */
-  constexpr double deltaAngle = pi * 20 / N;
+  const double deltaAngle = pi * 20 / N;
   double maxX = -1000.0;
   double minX = 1000.0;
   double maxY = -1000.0;
