diff -ruN VTK-9.3.1/Common/Core/SMP/Common/vtkSMPToolsImpl.h VTK-9.3.1.patched/Common/Core/SMP/Common/vtkSMPToolsImpl.h
--- VTK-9.3.1/Common/Core/SMP/Common/vtkSMPToolsImpl.h	2024-06-28 18:00:10.000000000 +0200
+++ VTK-9.3.1.patched/Common/Core/SMP/Common/vtkSMPToolsImpl.h	2024-08-15 13:35:52.611819716 +0200
@@ -43,7 +43,7 @@
 #endif
 
 template <BackendType Backend>
-class VTKCOMMONCORE_EXPORT vtkSMPToolsImpl
+class vtkSMPToolsImpl
 {
 public:
   //--------------------------------------------------------------------------------
@@ -59,7 +59,7 @@
   bool GetNestedParallelism() { return this->NestedActivated; }
 
   //--------------------------------------------------------------------------------
-  bool IsParallelScope() { return this->IsParallel; }
+  bool IsParallelScope();
 
   //--------------------------------------------------------------------------------
   bool GetSingleThread();
@@ -115,6 +115,12 @@
   std::atomic<bool> IsParallel{ false };
 };
 
+template <BackendType Backend>
+bool vtkSMPToolsImpl<Backend>::IsParallelScope()
+{
+  return this->IsParallel;
+}
+
 using ExecuteFunctorPtrType = void (*)(void*, vtkIdType, vtkIdType, vtkIdType);
 
 VTK_ABI_NAMESPACE_END
diff -ruN VTK-9.3.1/IO/EnSight/vtkEnSight6BinaryReader.cxx VTK-9.3.1.patched/IO/EnSight/vtkEnSight6BinaryReader.cxx
--- VTK-9.3.1/IO/EnSight/vtkEnSight6BinaryReader.cxx	2024-08-15 06:41:52.159851155 +0200
+++ VTK-9.3.1.patched/IO/EnSight/vtkEnSight6BinaryReader.cxx	2024-08-15 06:26:08.592479102 +0200
@@ -32,10 +32,13 @@
 #define VTK_STAT_STRUCT struct stat
 #define VTK_STAT_FUNC stat
 #else
-// here, we're relying on _FILE_OFFSET_BITS defined in vtkWin32Header.h to help
-// us on POSIX without resorting to using stat64.
+#if defined _LARGEFILE64_SOURCE
 #define VTK_STAT_STRUCT struct stat64
 #define VTK_STAT_FUNC stat64
+#else
+#define VTK_STAT_STRUCT struct stat
+#define VTK_STAT_FUNC stat
+#endif
 #endif
 
 VTK_ABI_NAMESPACE_BEGIN
diff -ruN VTK-9.3.1/IO/EnSight/vtkEnSightGoldBinaryReader.cxx VTK-9.3.1.patched/IO/EnSight/vtkEnSightGoldBinaryReader.cxx
--- VTK-9.3.1/IO/EnSight/vtkEnSightGoldBinaryReader.cxx	2024-08-15 06:41:52.159851155 +0200
+++ VTK-9.3.1.patched/IO/EnSight/vtkEnSightGoldBinaryReader.cxx	2024-08-15 06:26:20.599112124 +0200
@@ -38,10 +38,13 @@
 #define VTK_STAT_STRUCT struct stat
 #define VTK_STAT_FUNC stat
 #else
-// here, we're relying on _FILE_OFFSET_BITS defined in vtkWin32Header.h to help
-// us on POSIX without resorting to using stat64.
+#if defined _LARGEFILE64_SOURCE
 #define VTK_STAT_STRUCT struct stat64
 #define VTK_STAT_FUNC stat64
+#else
+#define VTK_STAT_STRUCT struct stat
+#define VTK_STAT_FUNC stat
+#endif
 #endif
 
 VTK_ABI_NAMESPACE_BEGIN
